src/repositories/fine.repository.ts

```diff
@@ -10,7 +10,7 @@ export class FinesRepository {
     return prisma.fine.delete({ where: { id } });
   }
 
-  async count(loanId: number) {
+  async countByLoanId(loanId: number) {
```

Comment:
This is okay, but it would be better if you could: 
1. Define a more common method for counting that accepts a params with the type of `Prisma.FineFindManyArgs`
2. This method would then call that common method with `loadId` as the argument.

i.e

```
count(params: Prisma.FineFindManyArgs) {
  return prisma.find.count(params)
}

countLoanById(loanId: number) {
  return this.count({ where: { loanId } })
}

```